lib/spack/spack/directory_layout.py

```diff
@@ -334,7 +335,7 @@ def path_for_spec(self, spec):
 
         path = self.relative_path_for_spec(spec)
         assert not path.startswith(self.root)
-        return os.path.join(self.root, path)
+        return str(pathlib.Path(self.root) / pathlib.Path(path))
```

Comment:
The value of `path` is not using the same path separator as `self.root`. os.path.join will append the path with windows appropriate separator but `path` uses the linux separator. We would get something like `C:\Users\...\test-debian6-core2/gcc-10.2.1/...`. We really only need the pathlib cast to switch the separators on the `path` variable and could rewrite it to `os.path.join(self.root, pathlib.Path(path))` if that's preferable.

Reply:
This makes sense, although I think it would also be worth updating `relative_path_for_spec` to return a string-based path with appropriate separators (and furthermore we might want to update `DirectoryLayout.__init__` to convert `self.root`.

Reply:
I moved the pathlib call into `relative_path_for_spec` when let me move the `path_for_spec` function back to no changes.